tests/Unit/DataStructures/Test_DataVector.py

```diff
@@ -81,14 +83,288 @@ def assignment_test():
+
+
+    def test_atan2(self):
+        a = DataVector(5, 0.67)
+        b = DataVector(5, -4.0)
+        self.assertEqual(a.atan2(b), DataVector(5, math.atan2(0.67, -4.0)))
+
+
+    def test_atanh(self):
+        a = DataVector(5, 0.67)
+        self.assertEqual(a.atanh(), DataVector(5, math.atanh(0.67)))
+        b = DataVector([0.0, -0.5])
+        self.assertEqual(b.atanh()[0], 0.0)
+        self.assertEqual(b.atanh()[1], math.atanh(-0.5))
+
+
+    def test_cbrt(self):
+        a = DataVector(5, 8.0)
+        self.assertEqual(a.cbrt(),DataVector(5, 2.0))
+        b = DataVector([1.0, 64.0])
+        self.assertEqual(b.cbrt()[0], 1.0)
+        self.assertEqual(b.cbrt()[1], 4.0)
+
+
+    def test_cos(self):
+        a = DataVector(5, 0.67)
+        self.assertEqual(a.cos(),DataVector(5, math.cos(0.67)))
+        b = DataVector([0.0, -0.5])
+        self.assertEqual(b.cos()[0], 1.0)
+        self.assertEqual(b.cos()[1], math.cos(-0.5))
+
+
+    def test_cosh(self):
+        a = DataVector(5, 1.0)
+        self.assertEqual(a.cosh(), DataVector(5, math.cosh(1.0)))
+        b = DataVector([0.0, -0.5])
+        self.assertEqual(b.cosh()[0], 1.0)
+        self.assertEqual(b.cosh()[1], math.cosh(-0.5))
+
+
+    def test_erf(self):
+        a = DataVector(5, 1.0)
+        self.assertAlmostEqual(a.erf()[0], math.erf(1.0))
+        b = DataVector([2.0, 3.0])
+        self.assertAlmostEqual(b.erf()[0], math.erf(2.0))
+        self.assertAlmostEqual(b.erf()[1], math.erf(3.0))
+
+
+    def test_erfc(self):
+        a = DataVector(5, 1)
+        self.assertAlmostEqual(a.erfc()[0], math.erfc(1.0))
+        b = DataVector([2.0, 3.0])
+        self.assertAlmostEqual(b.erfc()[0], math.erfc(2.0))
+        self.assertAlmostEqual(b.erfc()[1], math.erfc(3.0))
+
+
+    def test_exp(self):
+        a = DataVector(5, 1.0)
+        self.assertEqual(a.exp(), DataVector(5, math.exp(1.0)))
+        b = DataVector([2.0, -0.5])
+        self.assertEqual(b.exp()[0], math.exp(2.0))
+        self.assertEqual(b.exp()[1], math.exp(-0.5))
+
+
+    def test_exp2(self):
+        a = DataVector(5, 1.0)
+        self.assertEqual(a.exp2(), DataVector(5,2.0))
+        b = DataVector([2.0, -0.5])
+        self.assertAlmostEqual(b.exp2()[0], 4.0)
+        self.assertAlmostEqual(b.exp2()[1], 1.0 / math.sqrt(2.0))
+
+
+    def test_exp10(self):
+        a = DataVector(5, 1.0)
+        self.assertEqual(a.exp10(), DataVector(5, 10.0))
+        b = DataVector([2.0, -0.5])
+        self.assertEqual(b.exp10()[0], 100.0)
+        self.assertEqual(b.exp10()[1], 1.0 / math.sqrt(10.0))
+
+
+    def test_fabs(self):
+        a = DataVector(5, -0.67)
+        self.assertEqual(a.fabs(), DataVector(5, 0.67))
+        b = DataVector([2.0, -0.5])
+        self.assertEqual(b.fabs()[0], 2.0)
+        self.assertEqual(b.fabs()[1], 0.5)
+
+
+    def test_hypot(self):
+        a = DataVector(5, 3.0)
+        b = DataVector(5, 4.0)
+        self.assertEqual(a.hypot(b)[0], 5.0)
```

Comment:
Is there a reason not to have a different element test for `hypot` like you have for all the other functions?

Reply:
No, sorry, when I wrote all the other ones I was going to ask whether you wanted the tests for this and `atan2` to be two `DataVector`'s each with different elements or if you wanted only one to have differing elements but then I forgot.  I'll just add in the most general cases.